runelite-client/src/main/java/net/runelite/client/ui/NavigationButton.java

```diff
@@ -24,30 +25,52 @@
  */
 package net.runelite.client.ui;
 
-import java.awt.Image;
+import java.awt.image.BufferedImage;
 import java.util.function.Supplier;
-import javax.swing.ImageIcon;
-import javax.swing.JButton;
-import lombok.Getter;
-import lombok.extern.slf4j.Slf4j;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
 
-@Slf4j
-public class NavigationButton extends JButton
+/**
+ * UI navigation button.
+ */
+@Data
+@Builder
+@EqualsAndHashCode(of = {"name", "tooltip"})
+public class NavigationButton
 {
-	@Getter
-	private final Supplier<PluginPanel> panelSupplier;
-
-	public NavigationButton(String name, Image icon)
-	{
-		this(name, icon, null);
-	}
-
-	public NavigationButton(String name, Image icon, Supplier<PluginPanel> panelSupplier)
-	{
-		super();
-		setName(name);
-		setToolTipText(name);
-		setIcon(new ImageIcon(icon));
-		this.panelSupplier = panelSupplier;
-	}
+	/**
+	 * Button name.
+	 */
+	private final String name;
+
+	/**
+	 * Icon of button.
+	 */
+	private final BufferedImage icon;
+
+	/**
+	 * Tooltip to show when hovered.
+	 */
+	private String tooltip;
+
+	/**
+	 * Button selection state
+	 */
+	private boolean selected;
+
+	/**
+	 * On select action of the button.
+	 */
+	private Runnable onSelect;
+
+	/**
+	 * On click action of the button.
+	 */
+	private Runnable onClick;
+
+	/**
+	 * Supplier for plugin panel, used when expanding and contracting sidebar.
+	 */
+	private Supplier<PluginPanel> panel;
```

Comment:
None of the plugins dynamically create PluginPanels anymore, this should not be a supplier.

Reply:
Good point.